src/modules/common/db/core/DBProvider.tsx

```diff
@@ -8,7 +10,24 @@ interface DBProviderProps {
 const Context = createContext({})
 
 export const DBProvider: FC<DBProviderProps> = ({ children }) => {
-  const db = useDBConnection()
+  const { dataSource, isInitialized, error } = useDBConnection()
 
-  return <Context.Provider value={db}>{children}</Context.Provider>
+  logger.debug('debug')
+  logger.info('info')
+  logger.warn('warn')
+  logger.error('error')
+
+  if (error) {
+    logger.error('error', error)
+  }
+
+  if (!isInitialized) {
+    return (
+      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+        <ActivityIndicator size="large" color="#0000ff" />
+      </View>
+    )
+  }
```

Comment:
I think it should be problems with lint here

Reply:
If it is so important to leave the isInitialized state, it is better to change the logic of its operation to:
```suggestion
      try {
        setIsInitializing(true)
        logger.info('Initializing database...')
        await dataSource.initialize()
        setIsInitializing(false)
        logger.info('DataSource has been initialized!')
      } 
```
      
      if (isInitializing || error) return <Fullback />

Reply:
isInitialized is needed to display loading at application startup before the database is initialized